### PR TITLE
[#IOPID-1876] Enable Application Insights

### PR DESCRIPTION
**Pull request by gquadrati**
⚠️ This PR depends on new env variable

#### List of Changes

- Enable appinsights
- Add startup events
- Fix graceful shutdown

#### Motivation and Context

Enable logging on Session Manager

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/session-manager/env.example [hunk 1]
```diff
@@ -1,3 +1,12 @@
+################
+# AppInsinghts #
+################
+APPLICATIONINSIGHTS_CONNECTION_STRING=""
```
Comment (BurnedMarshal) [inline apps/session-manager/env.example:4]: is this `APPINSIGHTS_INSTRUMENTATIONKEY`?

File: apps/session-manager/src/server.ts [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+  AppInsightsConfig.APPLICATIONINSIGHTS_CONNECTION_STRING,
+  O.map((key) =>
+    initAppInsights(key, {
+      cloudRole: AppInsightsConfig.APPINSIGHTS_CLOUD_ROLE_NAME,
+      applicationVersion: getCurrentBackendVersion(),
+      disableAppInsights: AppInsightsConfig.APPINSIGHTS_DISABLED,
+      samplingPercentage: AppInsightsConfig.APPINSIGHTS_SAMPLING_PERCENTAGE,
+    }),
+  ),
+  O.getOrElseW(() => undefined),
```
Comment (BurnedMarshal) [inline apps/session-manager/src/server.ts:23]:
```suggestion
  O.toUndefined,
```

**Review by BurnedMarshal — COMMENTED** [review]
We are not binding winston with AppInsights because is not working on `io-backend`?
From io-backend
```
useWinstonFor({
  loggerId: LoggerId.event,
  transports: [withApplicationInsight(telemetryClient, "io-backend")],
})
```

File: apps/session-manager/env.example (hunk 1 above)
Comment (gquadrati) [reply apps/session-manager/env.example:4]: `APPINSIGHTS_INSTRUMENTATIONKEY` is now deprecated

**Comment by gquadrati** [comment]
> We are not binding winston with AppInsights because is not working on `io-backend`? From io-backend
> 
> ```
> useWinstonFor({
>   loggerId: LoggerId.event,
>   transports: [withApplicationInsight(telemetryClient, "io-backend")],
> })
> ```
> 
> Moreover is missing the graceful shutdown appinsights handling.

@BurnedMarshal this is needed for using `@pagopa/winston-ts`, but right now is not necessary

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**